FizzBuzz/src/test/java/rocks/kata/FizzBuzzTest.java

```diff
@@ -1,49 +1,145 @@
 package rocks.kata;
 
+import static org.junit.Assert.assertTrue;
 
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
 
 public class FizzBuzzTest {
 
-    private FizzBuzz fizzPattern = new FizzBuzzPattern();
-    private FizzBuzz fizzStringBuilder = new FizzBuzzStringBuilder();
-    private FizzBuzz fizzLiterals = new FizzBuzzLiterals();
-    private FizzBuzz fizzDivisibilityCheck = new FizzBuzzDivisibilityRules();
-
-    @ParameterizedTest
-    @ValueSource(ints = { 3, 12, 3 * 100003 })
-    void shouldReturnFizzForNumbersDivisibleByThree(int number) {
-        assertTrue(fizzPattern.returnFizzBuzzOrNumber(number).equals("Fizz"));
-        assertTrue(fizzStringBuilder.returnFizzBuzzOrNumber(number).equals("Fizz"));
-        assertTrue(fizzLiterals.returnFizzBuzzOrNumber(number).equals("Fizz"));
-        assertTrue(fizzDivisibilityCheck.returnFizzBuzzOrNumber(number).equals("Fizz"));
+    @TestTemplate
+    @ExtendWith(FizzBuzzTestInvocationContextProvider.class)
+    public void FizzBuzzTestTemplate(FizzBuzzTestCase testCase) {
```

Comment:
metody powinny zaczynać się małą literą + to, co jest w komentarzu poniżej mogłoby być zamiast tego nazwą metody